src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -3,10 +3,42 @@
 #include <unistd.h>
 
 #include <filesystem>
+#include <random>
 namespace fs = std::filesystem;
 
 namespace utl {
 
+namespace {
+std::string generate_unused_filename(const std::string& prefix,
+                                     const int length = 8)
+{
+  static const char alphanum[]
+      = "0123456789"
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        "abcdefghijklmnopqrstuvwxyz";
+  static const size_t alphanum_len = sizeof(alphanum) - 1;
+
+  std::random_device rd;
+  std::mt19937 generator(rd());
+  std::uniform_int_distribution<> dist(0, alphanum_len - 1);
+
+  std::string filename;
+  do {
+    std::ostringstream filename_stream;
+    filename_stream << prefix;
+    filename_stream << "--";
+
+    for (size_t i = 0; i < length; ++i) {
+      filename_stream << alphanum[dist(generator)];
```

Comment:
this could, in theory, cause an infinite loop if all single char possibilities are exhausted.

Why not use a simple counter instead, start at 1.

Reply:
It's 8 characters of 62 symbols so that would be 62^8 filenames available. Even with 100,000 files the chance of aliasing is 5e-10.

Reply:
Yes, I misread. Thanks!

But why write the code w random, why not just a counter?

Reply:
I can think of 2 reasons. One is that it reduces the possibility of race conditions. If you have two OR instances that both see file.odb.3 exists, they will both try to then write file.odb.4. A random approach makes aliasing nearly impossible.

The second is that a counter will linearly increase with number of tempfiles. So if file.odb.1 through file.odb.100 each exist, it will have to check 100 times before finding a name that doesn't exist.

The third and most likely reason is that this mirrors the implementation of `std::tmpnam()` without any compiler warnings.

Reply:
There cant be a race condition to write atomically to the same file, unless something is broken somewhere.

Reply:
This implementation has the same concern as tmpnam in that it checks for file existence and then opens the file.  It is possible, but improbable, to have something happen in between.   The other APIs directly return an open file to avoid that but don't play well with iostreams.

Reply:
I dont want to hold up anything here. I think this code is harmless even if it needs fixing or tweaks and fixing it if it is wrong is easy.